src/guildenstern/ctxfull.nim

```diff
@@ -70,7 +70,7 @@ proc receiveHttp(): bool {.gcsafe, raises:[] .} =
     let ret = recv(posix.SocketHandle(ctx.socketdata.socket), addr request[ctx.requestlen], expectedlength - ctx.requestlen, recvFlags)
     if ctx.requestlen == 0 and ret == 0:
       # connection closed
```

Comment:
`unregister()` is no longer needed by client code, this was an earlier partial fix to the problem but `closeSocket()` is the right thing to do here.